codecparsers/mpeg2_parser.cpp

```diff
@@ -568,36 +633,22 @@ namespace MPEG2 {
         bitReaderReadBits(10, &(m_sequenceHdr.vbv_buffer_size_value));
         bitReaderReadFlag(&(m_sequenceHdr.constrained_params_flag));
 
-        bitReaderReadFlag(&(m_sequenceHdr.load_intra_quantiser_matrix));
-
-        if (m_sequenceHdr.load_intra_quantiser_matrix) {
-            // read 8 bits *64 from the stream
-            uint32_t value;
-            for (uint8_t i(0); i < 64; ++i) {
-                bitReaderReadBits(8, &value);
-                m_sequenceHdr.intra_quantiser_matrix[i] = value;
-            }
-        } else {
+        readQuantMatrix(quantMatrices->load_intra_quantiser_matrix,
+                        quantMatrices->intra_quantiser_matrix);
+        if (!quantMatrices->load_intra_quantiser_matrix) {
             // use kDefaultIntraBlockMatrix
-            memcpy(m_sequenceHdr.intra_quantiser_matrix,
+            memcpy(quantMatrices->intra_quantiser_matrix,
                    &kDefaultIntraBlockMatrix[0],
                    kDefaultIntraBlockMatrix.size());
-            m_sequenceHdr.load_intra_quantiser_matrix = true;
+            quantMatrices->load_intra_quantiser_matrix = true;
         }
 
-        bitReaderReadFlag(&(m_sequenceHdr.load_non_intra_quantiser_matrix));
-
-        if (m_sequenceHdr.load_non_intra_quantiser_matrix) {
-            // read 8 bits * 64 from the stream
-            uint32_t value;
-            for (uint8_t i(0); i < 64; ++i) {
-                bitReaderReadBits(8, &value);
-                m_sequenceHdr.non_intra_quantiser_matrix[i] = value;
-            }
-        } else {
+        readQuantMatrix(quantMatrices->load_non_intra_quantiser_matrix,
```

Comment:
do you think it's ok to add other readQuantMatrix function with 3 params.
readQuantMatrix(bool& loadMatrix, uint8_t matrix[], uint8_t defaultMatirx[]).
 it will reduce your logical code.